### PR TITLE
test: integration test for get Financial Transactions By Wallet ID #183

### PR DESCRIPTION
**Pull request by poneciak57**
done:
- test when wallet exists and has financial transaction connected
- test when wallet does not exist

### CODE REVIEW

**Comment by poneciak57** [comment]
second test that checks when wallet id is incorrect fails because of the issue #163

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public void testGetFinancialTransactionsByWalletID_whenWalletIdIsIncorrect_thenReturnErrorResponseDTO() throws Exception {
+        mockMvc.perform(get("/api/transactions")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .queryParam("walletId", "1"))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+        Assertions.assertEquals(0, financialTransactionRepository.count());
+        Assertions.assertEquals(0, walletRepository.count());
+    }
+
+    private Wallet createTestWallet() {
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:63]: **This method has a declaration:**
```java
    private Wallet createTestWallet() {
```

**And this one:**
```java
     public void clearDb(){
```
**Why is the formatting of the code in them different?**

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+public class GetFinancialTransactionByWalletIdTest extends BaseIntegrationTestIT {
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @BeforeEach
+    public void clearDb(){
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:31]: Think (I'm sure) name of the method is wrong. 
It is:
```java
clearDb
```

It should be:
```java
clearDB
```

Link:
[What Does Database (DB) Mean?](https://www.techopedia.com/definition/1185/database-db)

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    @Autowired
+    WalletRepository walletRepository;
+
+    @BeforeEach
+    public void clearTestDB() {
+        financialTransactionRepository.deleteAll();
+        walletRepository.deleteAll();
+    }
+
+    @Test
```
Comment (DawidStuzynski) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:36]: Please add also @DisplayName with proper description

**Review by dawciobiel — CHANGES_REQUESTED** [review]
changes required

**Review by odzio33 — DISMISSED** [review]
Looks good for me

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:20]: Please don't use asterisk `*` when importing!

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:21]: Please don't use asterisk `*` when importing!

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        FinancialTransaction financialTransaction = createTestFinancialTransaction(wallet);
+        mockMvc.perform(get("/api/transactions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .queryParam("walletId", String.valueOf(wallet.getId())))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(jsonPath("$.[0].id").value(financialTransaction.getId()))
+                .andExpect(jsonPath("$.[0].amount").value(financialTransaction.getAmount()))
+                .andExpect(jsonPath("$.[0].description").value(financialTransaction.getDescription()))
+                .andExpect(jsonPath("$.[0].type").value(financialTransaction.getType().name()))
+                .andExpect(jsonPath("$.[0].date").value(financialTransaction.getDate().toString()));
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:50]: Add one blank line before using assertions.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    @DisplayName("when wallet id is correct returns error response dto and has 404 status code")
+    public void testGetFinancialTransactionsByWalletID_whenWalletIdIsIncorrect_thenReturnErrorResponseDTO() throws Exception {
+        mockMvc.perform(get("/api/transactions")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .queryParam("walletId", "1"))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+        Assertions.assertEquals(0, financialTransactionRepository.count());
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:62]: Add one blank line before using assertions.

**Review by dawciobiel — DISMISSED** [review]
Changes required

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:1]: Rename the file to GetFinancialTransactionByWalletIdIT.java (IT suffix - integration test).

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    private Wallet createTestWallet() {
+        return walletRepository.save(new Wallet("test_wallet"));
+    }
+
+    private FinancialTransaction createTestFinancialTransaction(Wallet wallet) {
+        return financialTransactionRepository.save(FinancialTransaction.builder()
+                .wallet(wallet)
+                .amount(BigDecimal.valueOf(2.0))
+                .date(Instant.ofEpochMilli(0L))
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdTest.java:76]:
```suggestion
                .date(Instant.ofEpochSecond(1L))
```
The minimum timestamp value is 1 second.

**Comment by mirekgab** [comment]
Please update your branch and resolve conflicts.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdIT.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @Test
+    @DisplayName("when wallet id is correct returns error response dto and has 404 status code")
+    public void givenInvalidWalletId_whenGetFinancialTransactionsByWalletId_thenNotFoundStatusCode() throws Exception {
+        mockMvc.perform(get("/api/transactions")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .queryParam("walletId", "1"))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/GetFinancialTransactionByWalletIdIT.java:63]: From a business point of view, the status returned in the response body is important. Add a check that this status has the correct value W003.
```json
{
    "status": "W003",
    "message": "WALLET_NOT_FOUND",
    "description": "Wallet with id: 38 does not exist",
    "statusCode": 404
}
```

File: src/main/resources/db/dbchangelog-test.yaml
Comment (mirekgab) [inline src/main/resources/db/dbchangelog-test.yaml:1]: Did you change anything in this file?

Comment (poneciak57) [reply src/main/resources/db/dbchangelog-test.yaml:1]: yes i changed it at the begining and now i only merged changes from main while resolving conflict

**Review by mirekgab — APPROVED** [review]
I think it looks good.

**Review by odzio33 — APPROVED** [review]
Looks good to me

**Review by Krisu93 — COMMENTED** [review]
Looks good for me

**Pull request merged**